platform.md

```diff
@@ -871,13 +873,15 @@ For more information on build reproducibility see [https://reproducible-builds.o
 ```toml
 [image]
```

Comment:
```suggestion
stack-id = "<stack ID>"
os = "<operating system>"
arch = "<architecture>"

[image]
```
I think we should add these keys to the top level as part of improved stack compat validation. These values should be the same for the previous image and the run image. The build should either fail if these don't match, or treat this like a first build and only provide `[run-image]` and not `[image]` in `analyzed.toml`.

You could make the argument that `os` in unnecessary here because that information is contained in the stack ID. I could buy that argument and leave `os` out.

Reply:
`detector` should fail if `stack-id`, `os` and `arch` don't match the build environment. If we added these keys to `stack.toml` `analyzer` could fail the build even earlier in this case, but right now `stack.toml` isn't required in analyzer, if `-run-image` is provided, and I don't want to make that file a hard requirement in phases that may not run on the builder image.

Reply:
Addressed in ea3add8 of #172

Reply:
I added the validation of `os`, `arch`, etc in c4b5746 of #172